<!-- -*- mode: XML -*- -->
<launch>

    ################ set up laser data ################
    <include file="$(find ugv_navigation)/launch/velodyne.launch"/>

    ################ start ugvdriver ################
    <include file="$(find ugv_navigation)/launch/ugvdriver.launch">
        <arg name="serial_port" value="/dev/ttyUSB1"/>
    </include>

    ################ publish an example base_link -> laser transform ################
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args="0.1 0 0 -1.28 0 0 /base_link /velodyne 100" />

    ################ draw path ################
    <include file="$(find ugv_navigation)/launch/path.launch">
	<arg name="fixed_frame" value="map"/>
        <arg name="use_odom" value="true"/>
    </include>

   ################ start gmapping ################
    <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
	<param name="base_frame" value="base_link"/>
	<param name="map_frame" value="map"/>
	<param name="odom_frame" value="odom"/>
	<param name="map_update_interval" value="1.0"/>
	<param name="linearUpdate" value="0.1"/>
	<param name="angularUpdate" value="0.1"/>
	<param name="maxUrange" value="130.0"/>
	<param name="sigma" value="0.05"/>
	<param name="kernelSize" value="1"/>
	<param name="lstep" value="0.05"/>
	<param name="astep" value="0.05"/>
	<param name="iterations" value="5"/>
	<param name="lsigma" value="0.075"/>
	<param name="ogain" value="3.0"/>
	<param name="lskip" value="0"/>
	<param name="srr" value="0.1"/>
	<param name="srt" value="0.2"/>
	<param name="str" value="0.1"/>
	<param name="stt" value="0.2"/>
	<param name="temporalUpdate" value="3.0"/>
	<param name="resampleThreshold" value="0.5"/>
	<param name="particles" value="30"/>
	<param name="xmin" value="-50.0"/>
	<param name="ymin" value="-50.0"/>
	<param name="xmax" value="50.0"/>
	<param name="ymax" value="50.0"/>
	<param name="delta" value="0.05"/>
	<param name="llsamplerange" value="0.01"/>
	<param name="llsamplestep" value="0.01"/>
	<param name="lasamplerange" value="0.005"/>
	<param name="lasamplestep" value="0.005"/>
    </node>

    ################ start move_base ################
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
	<param name="controller_frequency" value="10.0"/>
	<param name="controller_patiente" value="15.0"/>
	<rosparam file="$(find ugv_navigation)/params/slam_navigation/costmap_common_params.yaml" command="load" ns="global_costmap" />
	<rosparam file="$(find ugv_navigation)/params/slam_navigation/costmap_common_params.yaml" command="load" ns="local_costmap" />
	<rosparam file="$(find ugv_navigation)/params/slam_navigation/local_costmap_params.yaml" command="load" />
	<rosparam file="$(find ugv_navigation)/params/slam_navigation/global_costmap_params.yaml" command="load" />
	<rosparam file="$(find ugv_navigation)/params/slam_navigation/base_local_planner_params.yaml" command="load" />
    </node>

    ################ start goal ################
    <node pkg="ugv_navigation" type="goal_test_node" name="goal_test_node" output="screen">
	<param name="fix_frame" value="map"/>
	<param name="executetime" value="120"/>
	<param name="sleeptime" value="2"/>
	<param name="num_target" value="5"/>
	<rosparam param="P1">[5, -1.0, 0]</rosparam>
	<rosparam param="P2">[4.0, 3, 0]</rosparam>
	<rosparam param="P3">[1.0, 1.0, 0]</rosparam>
	<rosparam param="P4">[3.0, 0.0, 0]</rosparam>
	<rosparam param="P5">[0.0, 0.0, 0]</rosparam>
    </node>

    <!--<include file="$(find ugv_navigation)/launch/recvtarget.launch">
	<arg name="serial_port" value="/dev/ttyUSB0"/>
	<arg name="serial_baudrate" value="115200"/>
	<arg name="fix_frame" value="map"/>
	<arg name="executetime" value="1000"/>
	<arg name="sleeptime" value="5"/>
    </include>-->


    ################ start rviz ################
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ugv_navigation)/rviz/slam_navigation.rviz"/>

</launch>
